File: OpenMP_cpu/simd/2_safelen/solutions/.tacc/safelen.c
#include <math.h>
#define N 10000
double ac[N][N], as[N][N];

void  transform(int j,int offset){
//  offset will always be 16 or greater
int i;

   #pragma omp simd safelen(16)
   for(i=0;i<N;i++){
      as[j][i] = as[j][i-offset] + sin(as[j][i]);
      ac[j][i] = ac[j][i+offset] + cos(ac[j][i]);
   }
}

File: legacy/OpenMP_cpu/simd/2_safelen/solutions/.tacc/safelen.f90

subroutine transform(as,ac, j,offset)

integer,parameter :: N=1000000
integer           :: j,offset
double precision  :: as(n,n),ac(n,n)
!  offset will always be 16 or greater
integer :: i

   !$omp simd safelen(16)
   do i = offset, N-offset
      as(i,j) = as(i-offset,j) + sin(as(i,j))
      ac(i,j) = ac(i+offset,j) + cos(ac(i,j))
   end do

end subroutine
